### PR TITLE
951 potentially insecure hkdf use

### LINKED ISSUE #951: Potentially insecure hkdf use
## Current Behavior
`kdf := hkdf.New(hash, secret, nil, nil)`. No salt, no app-specific info.

## Expected Behavior
App-specific info and salt used.

## Possible Solution
Hardcode info, add some salt. Breaking change, but the gateway is not used in production.

## Your Environment
* Version of the product used: 0.30.0

### PR DESCRIPTION
**Pull request by smallhive**
Closes #951.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ run:
 # output configuration options
 output:
   # colored-line-number|line-number|json|tab|checkstyle|code-climate, default is "colored-line-number"
-  format: tab
+  formats: tab
```
Comment (cthulhu-rider) [inline .golangci.yml:15]: sure this is correct? maybe
```suggestion
  formats:
    - format: tab
```
https://golangci-lint.run/usage/configuration/

Comment (smallhive) [reply .golangci.yml]: You are right, fixed

**Review by cthulhu-rider — DISMISSED** [review]

**Comment by cthulhu-rider** [comment]
@smallhive i u already touch linter config, lets also do https://github.com/nspcc-dev/neofs-node/pull/2893/commits/cadd6b1b0b47c1de6912b0d68e9cfc39af35f091

**Review by roman-khimov — CHANGES_REQUESTED** [review]

File: creds/accessbox/accessbox.go [hunk 2]
```diff
@@ -19,6 +19,11 @@ import (
 	"google.golang.org/protobuf/proto"
 )
 
+var (
+	hkdfSalt = []byte{0x34, 0xf9, 0xe4, 0xfb, 0xaf, 0x1c, 0x7d, 0x6e}
```
[the remark this answers is not in the document]
Comment (smallhive) [reply creds/accessbox/accessbox.go]: Updated

File: creds/accessbox/accessbox.go [hunk 3]
```diff
@@ -19,6 +19,14 @@ import (
 	"google.golang.org/protobuf/proto"
 )
 
+const (
+	hkdfSaltLength = 8
```
Comment (roman-khimov) [inline creds/accessbox/accessbox.go:23]: Let's use 16, even though 8 is kinda sufficient, most people take more like https://en.wikipedia.org/wiki/PBKDF2

File: creds/accessbox/accessbox.go [hunk 4]
```diff
@@ -276,30 +289,31 @@ func encrypt(owner *keys.PrivateKey, sender *keys.PublicKey, data []byte) ([]byt
 		return nil, fmt.Errorf("generate random nonce: %w", err)
 	}
 
-	return enc.Seal(nonce, nonce, data, nil), nil
+	return append(hkdfSalt, enc.Seal(nonce, nonce, data, nil)...), nil
 }
 
 func decrypt(owner *keys.PrivateKey, sender *keys.PublicKey, data []byte) ([]byte, error) {
-	dec, err := getCipher(owner, sender)
+	dec, err := getCipher(owner, sender, data[:hkdfSaltLength])
```
Comment (roman-khimov) [inline creds/accessbox/accessbox.go:296]: Is `data` length checked anywhere?

Comment (cthulhu-rider) [inline creds/accessbox/accessbox.go:296]: panic potential

File: creds/accessbox/accessbox.go [hunk 5]
```diff
@@ -276,30 +289,31 @@ func encrypt(owner *keys.PrivateKey, sender *keys.PublicKey, data []byte) ([]byt
...
 	if err != nil {
 		return nil, fmt.Errorf("get chiper: %w", err)
 	}
+	origData := data[hkdfSaltLength:]
```
Comment (cthulhu-rider) [inline creds/accessbox/accessbox.go:300]:
```suggestion
	data = data[hkdfSaltLength:]
```

File: creds/accessbox/accessbox.go (hunk 3 above)
Comment (smallhive) [reply creds/accessbox/accessbox.go]: Ok

File: creds/accessbox/accessbox.go (hunk 4 above)
Comment (smallhive) [reply creds/accessbox/accessbox.go:296]: I didn't find such place

File: creds/accessbox/accessbox.go (hunk 5 above)
Comment (smallhive) [reply creds/accessbox/accessbox.go]: Updated

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**